Assets/GothicVR/Scripts/World/DijkstraPathFinder.cs

```diff
@@ -0,0 +1,308 @@
+        void OnDrawGizmos()
+        {
+            // Draw a yellow sphere at the transform's position
+            Gizmos.color = Color.green;
+            if (waypointsPosition == null)
+            {
+                return;
+            }
+            Gizmos.DrawLineList(waypointsPosition.ToArray());
+            if (_path != null)
+            {
+                var path = _path.Select(waypoint => waypoint._position).ToList();
+                var finalPath = new List<Vector3>();
+
+                for (int i = 0; i < path.Count; i++)
+                {
+                    finalPath.Add(path[i]);
+                    if (i != 0 && i != path.Count - 1)
+                    {
+                        finalPath.Add(path[i]);
+                    }
+                }
+                Gizmos.color = Color.red;
+                Gizmos.DrawLineList(finalPath.ToArray());
+            }
+
+        }
+
+        private void LightUpWaypoint(string wayPointName)
+        {
+            var waypoint = FindWaypointGO(wayPointName);
+            if (waypoint == null)
+            {
+                return;
+            }
+            waypoint.GetComponent<Renderer>().material.color = Color.green;
+        }
+
+        private GameObject FindWaypointGO(string wayPointName)
+        {
+            var result = gameObject.FindChildRecursively(wayPointName);
+            if (result != null)
+            {
+                return result;
+            }
+            else
+            {
+                Debug.Log("Waypoint not found");
+                return null;
+            }
+        }
+
+        private void Awake()
+        {
+            if (Instance == null)
+            {
+                Instance = this;
+            }
+            else
+            {
+                Destroy(gameObject);
+            }
+        }
+
+        public void SetDijkstraWaypoints(Dictionary<string, DijkstraWaypoint> waypoints)
+        {
+            DijkstraWaypoints = waypoints;
+
+            foreach (var waypoint in DijkstraWaypoints)
+            {
+                var result = gameObject.FindChildRecursively(waypoint.Key).transform.position;
+                if (result != null)
+                {
+                    waypoint.Value._position = result;
+                }
+            }
+            Debug.Log("DijkstraWaypoints set");
+            CalculateNeighbourDistances();
+
+        }
+
+        public DijkstraWaypoint[] FindFastestPath(DijkstraWaypoint start = null, DijkstraWaypoint end = null)
+        {
+            if (start == null)
+            {
+                start = DijkstraWaypoints[this.start] ?? throw new ArgumentException("The specified start waypoint is not in the queue.");
+            }
+            if (end == null)
+            {
+                end = DijkstraWaypoints[this.end] ?? throw new ArgumentException("The specified end waypoint is not in the queue.");
+            }
+
+            var startDijkstraWaypoint = DijkstraWaypoints[start];
```

Comment:
Error here. You try to get the waypoint object with the key, which would have to be a string for this struct, but is actually a waypointObject.